### PR TITLE
increase the liveness probe delay for GCE e2e tests to avoid premature teardown

### LINKED ISSUE #70305: kube-apiserver restarted on boot, caused flake
For https://gubernator.k8s.io/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-new-master-upgrade-cluster/2572?log#log the cluster upgrade failed.

Looking at the failure, it seems to be an inopportune restart of apiserver.  The new master comes up, apiserver starts, but then it restarts.  The upgrade test hits it during the restart, it looks like.  Logs: https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-new-master-upgrade-cluster/2572/artifacts/bootstrap-e2e-master/kube-apiserver.log-20181024-1540382414.gz

The reason for the apiserver restart was that kube-apiserver failed its liveness probe https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-new-master-upgrade-cluster/2572/artifacts/bootstrap-e2e-master/kubelet.log:
`I1024 05:18:39.900870    1569 kuberuntime_manager.go:565] Container "kube-apiserver" ({"docker" "2a1b78c8c811fe0939404eacf2eb815ff787ea636e1b6920fa9186c4b181eb1e"}) of pod kube-apiserver-bootstrap-e2e-master_kube-system(b13798debb0d0cb04a19e77831d7cf26): Container failed liveness probe.. Container will be killed and recreated.`

The liveness probe for api server is hitting :8080, and indeed it was failing its health check (apiserver log again):

```
I1024 05:18:28.000676       1 wrap.go:42] GET /healthz: (36.143742ms) 500
goroutine 5739 [running]:
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/httplog.(*respLogger).recordStatus(0xc006619180, 0x1f4)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/httplog/httplog.go:204 +0xd2
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/httplog.(*respLogger).WriteHeader(0xc006619180, 0x1f4)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/httplog/httplog.go:183 +0x35
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.(*baseTimeoutWriter).WriteHeader(0xc0029ef840, 0x1f4)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:197 +0xaf
net/http.Error(0x7feed9dd2ce8, 0xc0070ebc98, 0xc00cea4000, 0x338, 0x1f4)
        /usr/local/go/src/net/http/server.go:1976 +0xda
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/healthz.handleRootHealthz.func1(0x7feed9dd2ce8, 0xc0070ebc98, 0xc00d23fc00)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/healthz/healthz.go:162 +0x3fe
net/http.HandlerFunc.ServeHTTP(0xc014103880, 0x7feed9dd2ce8, 0xc0070ebc98, 0xc00d23fc00)
        /usr/local/go/src/net/http/server.go:1964 +0x44
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*pathHandler).ServeHTTP(0xc006eb0d80, 0x7feed9dd2ce8, 0xc0070ebc98, 0xc00d23fc00)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux/pathrecorder.go:241 +0x510
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*PathRecorderMux).ServeHTTP(0xc00a8cef50, 0x7feed9dd2ce8, 0xc0070ebc98, 0xc00d23fc00)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux/pathrecorder.go:234 +0x8a
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.director.ServeHTTP(0x3e7a8de, 0xf, 0xc008697170, 0xc00a8cef50, 0x7feed9dd2ce8, 0xc0070ebc98, 0xc00d23fc00)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/handler.go:154 +0x661
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithAuthorization.func1(0x7feed9dd2ce8, 0xc0070ebc98, 0xc00d23fc00)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters/authorization.go:64 +0x4b4
net/http.HandlerFunc.ServeHTTP(0xc00a8d3240, 0x7feed9dd2ce8, 0xc0070ebc98, 0xc00d23fc00)
        /usr/local/go/src/net/http/server.go:1964 +0x44
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.WithMaxInFlightLimit.func1(0x7feed9dd2ce8, 0xc0070ebc98, 0xc00d23fc00)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters/maxinflight.go:160 +0x3ff
net/http.HandlerFunc.ServeHTTP(0xc0060e7a40, 0x7feed9dd2ce8, 0xc0070ebc98, 0xc00d23fc00)
        /usr/local/go/src/net/http/server.go:1964 +0x44
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithImpersonation.func1(0x7feed9dd2ce8, 0xc0070ebc98, 0xc00d23fc00)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters/impersonation.go:50 +0x1ed9
net/http.HandlerFunc.ServeHTTP(0xc00a8d3280, 0x7feed9dd2ce8, 0xc0070ebc98, 0xc00d23fc00)
        /usr/local/go/src/net/http/server.go:1964 +0x44
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithAudit.func1(0x7feed9dd2ce8, 0xc0070ebc98, 0xc00d23fc00)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters/audit.go:54 +0x645
net/http.HandlerFunc.ServeHTTP(0xc00a8d32c0, 0x7feed9dd2ce8, 0xc0070ebc98, 0xc00d23fc00)
        /usr/local/go/src/net/http/server.go:1964 +0x44
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithAuthentication.func1(0x7feed9dd2ce8, 0xc0070ebc98, 0xc00d23fb00)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters/authentication.go:81 +0x456
net/http.HandlerFunc.ServeHTTP(0xc0042099f0, 0x7feed9dd2ce8, 0xc0070ebc98, 0xc00d23fb00)
        /usr/local/go/src/net/http/server.go:1964 +0x44
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP.func1(0xc00f040540, 0xc00a91cbe0, 0x5cb0580, 0xc0070ebc98, 0xc00d23fb00)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:100 +0xb3
created by k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:96 +0x1b0

logging error output: "[+]ping ok\n[+]log ok\n[+]etcd ok\n[+]poststarthook/generic-apiserver-start-informers ok\n[+]poststarthook/start-apiextensions-informers ok\n[+]poststarthook/start-apiextensions-controllers ok\n[+]poststarthook/bootstrap-controller ok\n[-]poststarthook/rbac/bootstrap-roles failed: reason withheld\n[+]poststarthook/scheduling/bootstrap-system-priority-classes ok\n[+]poststarthook/ca-registration ok\n[+]poststarthook/start-kube-apiserver-informers ok\n[+]poststarthook/start-kube-apiserver-admission-initializer ok\n[+]poststarthook/start-kube-aggregator-informers ok\n[+]poststarthook/apiservice-registration-controller ok\n[+]poststarthook/apiservice-status-available-controller ok\n[+]poststarthook/apiservice-openapi-controller ok\n[+]poststarthook/kube-apiserver-autoregistration ok\n[+]autoregister-completion ok\nhealthz check failed\n"
 [curl/7.38.0 35.232.52.86:34956]
```

I'm guessing this is due to `poststarthook/rbac/bootstrap-roles failed: reason withheld`

Looking at the next run, I'm hypothesizing that RBAC updates just take too long, and so it's marginal as to whether the bootstrap roles complete in time.

/sig api-machinery
/kind failing-test

cc @jberkus

### PR DESCRIPTION
**Pull request by logicalhan**
**What type of PR is this?**

/kind flake

**What this PR does / why we need it**:

This PR increases the liveness delay for kube-apiserver for GCE e2e tests. @justinsb ran some analysis and it turns out we are hitting our failureThreshold 75% of the time in our e2e tests (which means kubelet restarts the kube-apiserver 75% of the time). This is causing instability in our test results.

**Which issue(s) this PR fixes**:
Fixes #70305

```release-note
NONE
```

### CODE REVIEW

**Comment by AishSundar** [comment]
/milestone v1.13
/priority important-soon

@logicalhan do you plan on getting this in by Code freeze for 1.13, which is this friday 11/16 @ 5pm PST?

**Comment by logicalhan** [comment]
> Are we planning to incorporate #70676 ?

What do you mean? It should be unrelated, though there are some similar aspects. Also, that PR is merged.

**Comment by mikedanese** [comment]
I mean are we planning to use what is implemented in #70676 in the kube-apiserver liveness probe?

**Comment by logicalhan** [comment]
> /milestone v1.13
> /priority important-soon
> 
> @logicalhan do you plan on getting this in by Code freeze for 1.13, which is this friday 11/16 @ 5pm PST?

@AishSundar, yes ideally we would get this in prior to code freeze. My reasoning is that we'd benefit with increased test stability on 1.13 patch releases.

File: cluster/gce/config-test.sh [hunk 1]
```diff
@@ -53,6 +53,8 @@ NODE_LOCAL_SSDS_EXT=${NODE_LOCAL_SSDS_EXT:-}
 NODE_ACCELERATORS=${NODE_ACCELERATORS:-""}
 REGISTER_MASTER_KUBELET=${REGISTER_MASTER:-true}
 KUBE_APISERVER_REQUEST_TIMEOUT=300
+# Increase initial delay for the apiserver liveness probe, to avoid prematurely tearing it down
+KUBE_APISERVER_LIVENESS_PROBE_INITIAL_DELAY_SEC=${KUBE_APISERVER_LIVENESS_PROBE_INITIAL_DELAY_SEC:-45}
```
Comment (cheftako) [inline cluster/gce/config-test.sh]: As long as ETCD liveness is part of the KAS healthz checks I would keep them in sync on the initial delay.

Comment (logicalhan) [reply cluster/gce/config-test.sh]: Sure, that makes sense. I've made the corresponding bump for etcd.

Comment (logicalhan) [reply cluster/gce/config-test.sh]: I verified the etcd configs on GCE deployed e2e cluster via manual inspection of the manifest files. Resolving this.

File: cluster/gce/config-test.sh [hunk 2]
```diff
@@ -53,6 +53,10 @@ NODE_LOCAL_SSDS_EXT=${NODE_LOCAL_SSDS_EXT:-}
...
+# Also increase the initial delay for etcd just to be safe
+ETCD_LIVENESS_PROBE_INITIAL_DELAY_SEC=${ETCD_LIVENESS_PROBE_INITIAL_DELAY_SEC:-45}
```
Comment (justinsb) [inline cluster/gce/config-test.sh]: Yes, I think this makes sense - I can imagine the crash loops if e.g. we have a big etcd database.  Better safe than sorry!

**Comment by justinsb** [comment]
/lgtm

@logicalhan and I discussed this; there's probably something more correct we can do around the probes, which @logicalhan was starting in #70676.  But given the timing, this is the best option for 1.13 IMO.

(I also have a long standing wish for an HTTP probe that can be used by loadbalancers that returns a 200, but doesn't require enabling anonymous authentication, so maybe we can look at that at the same time in 1.14)

**Pull request merged**